.changeset/salty-pianos-sell.md

```diff
@@ -0,0 +1,7 @@
+---
+"ponder": patch
+---
+
+Improved retry behavior for `context.client.getBlock()`, `context.client.getTransaction()`, `context.client.getTransactionReceipt()`, and `context.client.getTransactionConfirmations()`. This fixes the error: `BlockNotFoundError: Block at number "[number]" could not be found`.
```

Comment:
Last sentence could be more specific e.g.

Now, the error `BlockNotFoundError: Block at number "[number]" could not be found` will be retried properly. This solves a common issue where RPCs return empty data for near-tip blocks.